app/index.html

```diff
@@ -5,6 +5,17 @@
 	<title>Lunaria</title>
 	<meta charset='utf-8' />
 	<meta name='viewport' content='width=device-width, initial-scale=1.0' />
+
+	<!-- support hosting from a subpath such as ipfs -->
+	<script>
+		const subPath = window.location.pathname.substring(0, window.location.pathname.lastIndexOf('/'))
```

Comment:
```ts
'/subfolder'.lastIndexOf('/') === 0
```
I don't think this will do what you want as it will strip the `subfolder`, but we want to keep it.

Reply:
```suggestion
		const subPath = window.location.pathname.endsWith('index.html')
			? window.location.pathname.substring(0, window.location.pathname.length - 'index.html'.length)
			: window.location.pathname
```
I think something like this may work?  Double check that a base href of '/subfolder' and '/subfolder/' work the same though, as this will not add a trailing slash if it is missing.

Reply:
I just made it so that it will always remove `index.html` if it's in the path and just take the subpath from out of it by removing the trailing slash. will re-add it later as needed by the base element.

```ts
const subPath = window.location.pathname.replace('index.html', '').replace(/\/$/, '')
```

<img width="608" alt="Screenshot 2024-01-18 at 12 50 40 PM" src="https://github.com/DarkFlorist/lunaria/assets/1169838/d26ee002-eeaf-4d11-87f8-def183f1c319">